src/scripts/items/BattleItem.ts

```diff
@@ -4,9 +4,31 @@ class BattleItem extends Item {
     type: GameConstants.BattleItemType;
 
     constructor(type: GameConstants.BattleItemType) {
-        let basePrice = 0;
+        let basePrice = 1000;
+
+        switch (type) {
+            case GameConstants.BattleItemType.xAttack:
+                basePrice = GameConstants.ItemPrice.xAttack;
+                break;
+            case GameConstants.BattleItemType.xClick:
+                basePrice = GameConstants.ItemPrice.xClick;
+                break;
+            case GameConstants.BattleItemType.xExp:
+                basePrice = GameConstants.ItemPrice.xExp;
+                break;
+            case GameConstants.BattleItemType.Token_collector:
+                basePrice = GameConstants.ItemPrice.Token_collector;
+                break;
+            case GameConstants.BattleItemType.Item_magnet:
+                basePrice = GameConstants.ItemPrice.Item_magnet;
+                break;
+            case GameConstants.BattleItemType.Lucky_incense:
+                basePrice = GameConstants.ItemPrice.Lucky_incense;
+                break;
```

Comment:
Could this be simplified to `GameConstants.ItemPrice[ GameConstants.BattleItemType[type] ]` for every case?
Similarly for the other item types.